tests/test_marquez_client.py

```diff
@@ -258,6 +258,75 @@ def test_create_job(self, mock_put):
         assert str(response['id']) is not None
         assert str(response['location']) == location
 
+    @mock.patch("marquez_client.client.MarquezClient._put")
+    def test_create_job_with_run_id(self, mock_put):
+        run_id = str(uuid.uuid4())
+        job_name = "my-job"
+        input_dataset = [
+            {
+                "namespace": "my-namespace",
+                "name": "public.mytable"
+            }
+        ]
+        output_dataset = {
+            "namespace": "my-namespace",
+            "name": "public.mytable"
+        }
+
+        location = "https://github.com/my-jobs/blob/" \
+                   "07f3d2dfc8186cadae9146719e70294a4c7a8ee8"
+
+        context = {
+            "sql": "SELECT * FROM public.mytable;"
+        }
+
+        mock_put.return_value = {
+            "id": {
+                "namespace": "my-namespace",
+                "name": "my-job"
+            },
+            "type": "BATCH",
+            "name": "my-job",
+            "createdAt": "2020-08-12T07:30:55.321059Z",
+            "updatedAt": "2020-08-12T07:30:55.333230Z",
+            "namespace": "my-namespace",
+            "inputs": [
+                {
+                    "namespace": "my-namespace",
+                    "name": "public.mytable"
+                }
+            ],
+            "outputs": [
+                {
+                    "namespace": "my-namespace",
+                    "name": "public.mytable"
+                }
+            ],
+            "location": "https://github.com/my-jobs/blob/"
+                        "07f3d2dfc8186cadae9146719e70294a4c7a8ee8",
+            "context": {
+                "sql": "SELECT * FROM public.mytable;"
+            },
+            "description": "My first job.",
+            "latestRun": None,
+            "runId": run_id
+        }
+
+        response = self.client.create_job(
+            namespace_name=_NAMESPACE,
+            job_name=job_name,
+            job_type=JobType.BATCH,
+            location=location,
+            input_dataset=input_dataset,
+            output_dataset=output_dataset,
+            context=context,
+            run_id=run_id
+        )
+
+        assert str(response['id']) is not None
+        assert str(response['location']) == location
+        assert str(response['runId']) == run_id
```

Comment:
you need to check that mock_put was called with the parameters you expect.
Here is an example here: 
https://github.com/MarquezProject/marquez-airflow/blob/a5a72879cfa7b4a5db1bcd6b3a11fc942943405e/tests/test_marquez_dag.py#L97

you can also assert that response is mock_put.return_value since it should returning that.

Reply:
it makes sense in the dag as there are many calls happening. this method is directly calling only one.

return value already there few lines above:
https://github.com/MarquezProject/marquez-python/blob/abae9bc6c08c858f8d729eec527cfa4dc43b7164/tests/test_marquez_client.py#L283

Reply:
response = return value from create_job (inside return value of mock.put).
python magic.

Reply:
As mentioned in my comment above, you need to check that mock_put was called with the parameters you expect.
You want to verify that the `run_id` parameter passed to create_job is actually sent to the backend.
I would _assert mock_put.call_args.kwargs["runId"] == run_id_ (or something similar)
Doc here: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_args

Reply:
done.